changelog/12749.feature.rst

```diff
@@ -0,0 +1,3 @@
+Add :confval:`collect_imported_tests`, when enabled (default is disabled) will make sure to not consider classes/functions which are imported by a test file and contains Test/test_*/*_test.
```

Comment:
As per my comment, this description needs an update:

```suggestion
New :confval:`collect_imported_tests`: when enabled (the default) pytest will collect classes/functions in test modules even if they are imported from another file.

Setting this to True will make pytest collect classes/functions from test files only if they are defined in that file (as opposed to imported there).
```

Also, we should describe this option in https://github.com/pytest-dev/pytest/blob/main/doc/en/reference/reference.rst in a new `.. conf-val::` block.